pkg/apis/pxc/v1/pxc_types.go

```diff
@@ -717,7 +718,10 @@ func (cr *PerconaXtraDBCluster) Version() *v.Version {
 // CompareVersionWith compares given version to current version. Returns -1, 0, or 1 if given version is smaller, equal, or larger than the current version, respectively.
 func (cr *PerconaXtraDBCluster) CompareVersionWith(version string) int {
 	if len(cr.Spec.CRVersion) == 0 {
-		cr.setVersion()
+		_, err := cr.setVersion()
+		if err != nil {
+			log.Fatalf("failed to set version: %v", err)
```

Comment:
@spron-in @delgod 
Should we really die there?
Or maybe just use current operator version?

Reply:
I like this idea to add logic into setVersion function about using current operator version

Reply:
Why should we mix operator version with CR version in this function? Does it make any scenes? If it does why do we even need to compare versions in our code?
My approach if to fail fast before operator make any damage to cluster. Your approach is if we can't determine version of CR assume we are using same as operator.
Should we do it?

Reply:
But falling means that operator will be always restarting and probably do not process any other clusters

Reply:
I don't like the idea of crashing the operator here:

1. Operator might be managing other clusters which are not impacted by this error
2. We still need the operator to fix the cluster. What is the solution if it crashes?

Reply:
https://github.com/percona/percona-xtradb-cluster-operator/pull/803/commits/bcd8dde4a6de9c52102213840f828f55017c8d2c

Reply:
https://github.com/percona/percona-xtradb-cluster-operator/pull/803/commits/21d24816a6df26a51734b67e6e31c8f48343ced1